src/github.ts

```diff
@@ -266,23 +267,25 @@ export class GitHub {
     return this.graphql(opts);
   }
 
-  private async graphqlRequest(
-    opts: {
-      [key: string]: string | number | null | undefined;
-    },
-    maxRetries = 1
-  ) {
-    while (maxRetries >= 0) {
-      try {
-        return await this.makeGraphqlRequest(opts);
-      } catch (err) {
-        if (err.status !== 502) {
-          throw err;
+  private graphqlRequest = wrapAsync(
```

Comment:
Did you confirm that `wrapAsync` retains stack trace information from the original error?

You might be able to copy the `.stack` property from the Error you're wrapping on the new error object you're creating. Or, alternatively, fiddle with the [captureStackStrace](https://nodejs.org/api/errors.html#errors_error_capturestacktrace_targetobject_constructoropt) method.

Reply:
We're now capturing the original error in the wrapped error as `cause?: Error`.